### PR TITLE
Migrated to null safety version

### PR DESCRIPTION
**Pull request by trustmefelix**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -5,11 +5,13 @@ homepage: https://github.com/ALMaclaine/dart_string_stats
...
+  sdk: '>=2.12.0 <3.0.0'
 
 #dependencies:
 #  path: ^1.6.0
 
 dev_dependencies:
   test: ^1.6.0
   effective_dart: ^1.2.1
+  build_runner: ^1.10.1
+  pedantic: ^1.9.2
```
Comment (ALMaclaine) [inline pubspec.yaml:17]: Will you add a new line here please?

**Review by ALMaclaine — CHANGES_REQUESTED** [review]
Hey, thanks for the contribution.

Looks fine overall and plan to merge.

Before so, can you please address the following items:

1. Fix the comment in the code review
2. Give a short justification for the changes

File: pubspec.yaml (hunk 1 above)
Comment (trustmefelix) [reply pubspec.yaml]: Thanks for reminding, these packages were used on my own logic checking, and now have been removed.

**Comment by ALMaclaine** [comment]
@trustmefelix Hey I ran the tests today before publishing and one test is failing now:

00:01 +22 -1: test/string_stats_test.dart: processing_function_tests StatisticsCounter works [E]
  type '_CompactLinkedHashSet<Position?>' is not a subtype of type 'Set<Position>' of 'other'
  dart:collection                                                            SetMixin.union
  package:string_stats/src/string_statistics.dart 34:75                      StringStatistics.combine.<fn>.<fn>
  dart:collection                                                            MapMixin.update
  package:string_stats/src/string_statistics.dart 34:49                      StringStatistics.combine.<fn>
  dart:collection                                                            _LinkedHashMapMixin.forEach
  package:string_stats/src/string_statistics.dart 34:10                      StringStatistics.combine
  package:string_stats/src/processing_classes/statistics_counter.dart 31:22  StatisticsCounter.add
  test/tests/processing_class.tests.dart 58:26                               processing_class_tests.<fn>.<fn>

00:01 +25 -1: Some tests failed.

**Comment by trustmefelix** [comment]
> @trustmefelix Hey I ran the tests today before publishing and one test is failing now:
> 
> 00:01 +22 -1: test/string_stats_test.dart: processing_function_tests StatisticsCounter works [E]
> type '_CompactLinkedHashSet<Position?>' is not a subtype of type 'Set' of 'other'
> dart:collection SetMixin.union
> package:string_stats/src/string_statistics.dart 34:75 StringStatistics.combine..
> dart:collection MapMixin.update
> package:string_stats/src/string_statistics.dart 34:49 StringStatistics.combine.
> dart:collection _LinkedHashMapMixin.forEach
> package:string_stats/src/string_statistics.dart 34:10 StringStatistics.combine
> package:string_stats/src/processing_classes/statistics_counter.dart 31:22 StatisticsCounter.add
> test/tests/processing_class.tests.dart 58:26 processing_class_tests..
> 
> 00:01 +25 -1: Some tests failed.

@ALMaclaine The code has been updated and run the unit test, and all test passed. Please kindly help to review it.

**Comment by ALMaclaine** [comment]
I'm sorry, I don't understand what you mean.

**Comment by trustmefelix** [comment]
@ALMaclaine I just updated my branch for this bug fix.

**Pull request merged**